### PR TITLE
[VAULT-3472] Cap Client `id_token_ttl` field to associated Key's `verification_ttl`

### PR DESCRIPTION
**Pull request by vinay-gopalan**
This PR caps the `id_token_ttl` field for the Client, ensuring that it can't be greater than the `verification_ttl` of the Named Key associated with the Client. A test to confirm the behavior is also included.

```
$ vault write identity/oidc/key/a-key rotation_period=1m verification_ttl=1m allowed_client_ids="*"

$ vault % vault write identity/oidc/client/a-client \
> key="a-key" \
> id_token_ttl=3600

Error writing data to identity/oidc/client/a-client: Error making API request.

URL: PUT http://127.0.0.1:8200/v1/identity/oidc/client/a-client
Code: 400. Errors:

* a client's id_token_ttl cannot be greater than the verification_ttl of the key it references
```

### CODE REVIEW

File: vault/identity_store_oidc_provider_test.go [hunk 1]
```diff
@@ -972,6 +974,53 @@ func TestOIDC_Path_OIDC_ProviderClient_NilKeyEntry(t *testing.T) {
...
+	c.identityStore.HandleRequest(ctx, &logical.Request{
+		Path:      "oidc/key/test-key",
+		Operation: logical.CreateOperation,
+		Data: map[string]interface{}{
+			"verification_ttl": int64(60),
+		},
+		Storage: storage,
+	})
+
+	// Create a test role "test-client" with an id_token_ttl longer than the
```
Comment (fairclothjm) [inline vault/identity_store_oidc_provider_test.go:993]: nit: comment should say `Create a test client`

**Review by fairclothjm — CHANGES_REQUESTED** [review]
Looks good so far! We should also consider the case where a `client` and `key` already exist and we get an update on the key's `verification_ttl`.

**Review by austingebauer — COMMENTED** [review]
Thanks, @vinay-gopalan! Looks good so far. No other comments besides a +1 to @fairclothjm suggestion around checking at key update time.

File: vault/identity_store_oidc_provider_test.go (hunk 1 above)
Comment (vinay-gopalan) [reply vault/identity_store_oidc_provider_test.go]: Updated

**Comment by vinay-gopalan** [comment]
Thanks for the review @fairclothjm and @austingebauer ! I made the updates to the Key update method and also included a test

**Review by fairclothjm — APPROVED** [review]

**Review by austingebauer — APPROVED** [review]

**Pull request merged**